word_search.py

```diff
@@ -0,0 +1,102 @@
+'''
+Created on 9 June 2017
+
+@author: Eva
+'''
+from queue import Queue
+
+class WordSearch:
```

Comment:
Yes, the order of rows and columns is always arbitrary in a 2-d array.

One thing you can do sometimes to help there is to wrap your underlying data structure, and create an API that is harder to use incorrectly.

Here is an example that probably is a bit contrived but helps explain the concept:
https://github.com/flerdacodeu/codeu/blob/api/api/matrix.py